File: HackerEarth/DreamInCode/cake walk++.py
'''
1
5 3
1 2 3 4 5
1 3
3 4
5 1
'''

for i in range(int(input())):
    print("Case {}:".format(i+1))

    students, updates = [int(x) for x in input().split()]
    scores = [int(x) for x in input().split()]
    flag = 0
    for i in range(1, students):
        if scores[i] != scores[i-1]:flag = 1
    for _ in range(updates):
        x, y = [int(x) for x in input().split()]
        if not flag:
            print(0)
            continue
        if scores[0] == x: scores[0] = y
        sums = 0
        for i in range(1, students):
            if scores[i] == x:scores[i] = y
            sums += 2 * (abs(scores[i] - scores[i-1]))
        print(sums//2)

File: HackerEarth/DreamInCode/cake_walk++.go
package main

import (
	"fmt"
)

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func main() {
	var caseno, students, updates, x, y, sums int
	fmt.Scan(&caseno)
	for i := 0; i < caseno; i++ {
		fmt.Printf("Case %d:", caseno)
		fmt.Println()
		fmt.Scan(&students)
		fmt.Scan(&updates)
		flag := 0
		scores := make([]int, students+1)
		fmt.Scan(&scores[0])
		for j := 1; j < students; j++ {
			fmt.Scan(&scores[j])
			if scores[j] != scores[j-1] {
				flag = 1
			}
		}
		for j := 0; j < updates; j++ {
			fmt.Scan(&x)
			fmt.Scan(&y)
			if flag == 0 {
				fmt.Println(0)
			}
			if scores[0] == x {
				scores[0] = y
			}
			sums = 0
			for a := 1; a < students; a++ {
				if scores[a] == x {
					scores[a] = y
				}
				sums += 2 * (abs(scores[a] - scores[a-1]))
			}
			fmt.Println(int(sums / 2))
		}

	}

}
